c2corg_ui/templates/waypoint/edit.html

```diff
@@ -1,23 +1,94 @@
-  <li>Name: <input type="text" value="${waypoint_title}" /></li>
-  <li>Elevation: <input type="text" value="${waypoint['elevation']}" /> m</li>
-</ul>
+<h1>${self.pagetitle()}</h1>
+<form app-document-editing="waypoints" app-document-editing-model="waypoint"
+% if updating_doc:
+  app-document-editing-id="${waypoint_id}" app-document-editing-culture="${waypoint_culture}"
+% endif
+  name="editForm" novalidate ng-submit="editCtrl.submitForm(editForm.$valid)">
+  <div id="culture-group" class="form-group" ng-class="{ 'has-error': editForm.culture.$touched && editForm.culture.$invalid }">
+    <label>Culture</label>
+    % if waypoint_culture:
+      ${waypoint_culture}
+    % else:
+      <select name="culture" ng-options="culture for culture in ['${"','".join(available_cultures) |n}']" 
+        ng-model="waypoint.locales[0].culture" class="form-control" required>
+      </select>
+      <div class="help-block" ng-messages="editForm.culture.$error" ng-if="editForm.culture.$touched">
+        <p ng-message="required">Culture is required.</p> 
+      </div>
+    % endif
+  </div>
+  <div id="title-group" class="form-group" ng-class="{ 'has-error': editForm.title.$touched && editForm.title.$invalid }">
+    <label>Title</label>
+    <input type="text" name="title" ng-model="waypoint.locales[0].title" class="form-control" required ng-minlength="3" />
+    <div class="help-block" ng-messages="editForm.title.$error" ng-if="editForm.title.$touched">
+      <p ng-message="required">Title is required.</p> 
+      <p ng-message="minlength">Title is too short.</p>
+    </div>
+  </div>
+  <div id="waypoint_type-group" class="form-group" ng-class="{ 'has-error': editForm.waypoint_type.$touched && editForm.waypoint_type.$invalid }">
+    <label>Type</label>
+    <select name="waypoint_type" ng-options="type for type in ['${"','".join(waypoint_types) |n}']"
+      ng-model="waypoint.waypoint_type" class="form-control" required>
+    </select>
+    <div class="help-block" ng-messages="editForm.waypoint_type.$error" ng-if="editForm.waypoint_type.$touched">
+      <p ng-message="required">Type is required.</p> 
+    </div>
+  </div>
+  <div id="elevation-group" class="form-group" ng-class="{ 'has-error': editForm.elevation.$touched && editForm.elevation.$invalid }">
+    <label>Elevation</label>
+    <input type="number" name="elevation" ng-model="waypoint.elevation" class="form-control" required ng-maxlength="4" /> m
+    <div class="help-block" ng-messages="editForm.elevation.$error" ng-if="editForm.elevation.$touched">
+      <p ng-message="required">Elevation is required.</p> 
+      <p ng-message="maxlength">Elevation must be smaller than 9999 m.</p>
+    </div>
+  </div>
+  <div id="longitude-group" class="form-group" ng-class="{ 'has-error': editForm.longitude.$touched && editForm.longitude.$invalid }">
+    <label>Longitude</label>
+    <input type="number" name="longitude" ng-model="waypoint.geometry.geom.longitude" class="form-control" required /> °E
```

Comment:
What about setting the `min`/`max` attributes for lon/lat? Also for elevation.

    <input type="number" name="input" ng-model="example.value"
           min="0" max="99" required>

See: https://docs.angularjs.org/api/ng/input/input%5Bnumber%5D